app/src/forms/form/service.js

```diff
@@ -405,7 +406,76 @@ const service = {
       throw err;
     }
   },
+  createMultiSubmission: async (formVersionId, data, currentUser) => {
+    let trx;
+    try {
+      const formVersion = await service.readVersion(formVersionId);
+      const { identityProviders } = await service.readForm(formVersion.formId);
+
+      trx = await FormSubmission.startTransaction();
+
+      // Ensure we only record the user if the form is not public facing
+      const isPublicForm = identityProviders.some((idp) => idp.code === 'public');
+      const createdBy = isPublicForm ? 'public' : currentUser.usernameIdp;
```

Comment:
Public forms are not allowed to upload files. You need to prevent users from being able to upload if they aren't logged in. It should never get to this point.

Reply:
We solve it

Reply:
Done